### ISSUE TITLE
[consistent-type-imports]It reports error when I import a `data` and only use `typeof data`.

### ISSUE THREAD

**Issue by drylint** [typescript-eslint/typescript-eslint#4666]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/consistent-type-imports": ["error"]
  }
}
```

```TS
// a.ts
export const data = {
  num: 1,
  str: 'hello',
}
```

```ts
// b.ts
import { data } from './a'

export const fn = (params: typeof data): void => {
  // ...
}

```

**Expected Result**

I expect that is no error.

**Actual Result**

It reports `All imports in the declaration are only used as types. Use `import type`.`

<img width="905" alt="image" src="https://user-images.githubusercontent.com/80906525/157672429-622e7613-a9dd-4514-b226-7fdf2c497484.png">

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `5.13.0` |
| `@typescript-eslint/parser`        | `5.13.0` |
| `TypeScript`                       | `4.5.5` |
| `ESLint`                           | `8.10.0` |
| `node`                             | `16.13.1` |

**Comment by Josh-Cena** [comment]
You can of course change it to `import type { data } from "./a"`, because all that does is to tell the transpiler to remove the import statement here, which of course can be removed, because after stripping `typeof data`, there're no longer references to `data`.

**Comment by JoshuaKGoldberg** [comment]
Yup! This is what the rule is meant to do. A `: typeof data` type annotation is purely within the type system. `data` is not going to be used in the transpiled JavaScript code.

You can preview what the JS equivalent for any TS syntax is -regardless of type errors and/or lint complaints- on the TypeScript playground: https://www.typescriptlang.org/play?ssl=3&ssc=29&pln=3&pc=28#code/JYWwDg9gTgLgBAbzgEwIY1XAvnAZlCEOAcgDoB6VYgWACg6BTAD0ljgGMIA7AZ3ly5wAvHAAUYVFFQgeALjgwAnmAYRcKdKgCU8gG4RgyYQD5EdOHHLk4pW3Sx0gA
